### PR TITLE
Add previously written methods to challenge seed in "Remove an Element from a Max Heap"

### PR DESCRIPTION
**Pull request by TyMick**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

[Remove an Element from a Max Heap](https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/remove-an-element-from-a-max-heap) currently tells students to add the `insert` method they just wrote in the [previous lesson](https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/insert-an-element-into-a-max-heap) to their code in this lesson as well, but unless the student made an effort to save their previous code, they can't go back and copy it, in which case they have to rewrite it from scratch.

This change copies the methods over from the previous lesson's solution to this lesson's challenge seed so students don't have to needlessly duplicate effort.

I also made a couple quick changes to the previous lesson's solution to bring it in line with its instructions (making the starting heap `[null]` instead of `[undefined]` and only returning *inserted* elements in the `print` method).

### CODE REVIEW

**Comment by moT01** [comment]
I have blocked this for now. See the [pinned code freeze issue](https://github.com/freeCodeCamp/freeCodeCamp/issues/38308)

**Comment by raisedadead** [comment]
Limitations removed on Coding Interview Prep Challenges.

**Review by tinktink1 — DISMISSED** [review]
I have reviewed your changes and they look good to me. I think your instructions update are good! Your code is also formatted well.

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.english.md [hunk 1]
```diff
@@ -68,20 +68,20 @@ var MaxHeap = function() {
...
+		while (ele > arr[Math.floor(index / 2)] && index > 1) {
 			arr[index] = arr[Math.floor(index / 2)];
 			arr[Math.floor(index / 2)] = ele;
 			index = arr[Math.floor(index / 2)];
 		}
 		this.heap = arr;
 	}
 	this.print = () => {
-		return this.heap
+		return this.heap.slice(1);
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.english.md:84]:
```suggestion
		return this.heap;
```

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.english.md [hunk 2]
```diff
@@ -68,20 +68,20 @@ var MaxHeap = function() {
 ```js
 var MaxHeap = function() {
 	// change code below this line
-	this.heap = [undefined];
+	this.heap = [null];
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.english.md:71]:
```suggestion
	this.heap = [];
```

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.english.md [hunk 3]
```diff
@@ -68,20 +68,20 @@ var MaxHeap = function() {
...
 	this.insert = (ele) => {
 		var index = this.heap.length;
 		var arr = [...this.heap];
 		arr.push(ele);
-		while (ele > arr[Math.floor(index / 2)]) {
+		while (ele > arr[Math.floor(index / 2)] && index > 1) {
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.english.md:76]:
```suggestion
		while (ele > arr[Math.floor(index / 2)]) {
```

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md [hunk 4]
```diff
@@ -46,6 +46,21 @@ tests:
 
 ```js
 var MaxHeap = function() {
+  this.heap = [null];
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md:49]:
```suggestion
  this.heap = [];
```

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md [hunk 5]
```diff
@@ -46,6 +46,21 @@ tests:
...
+  this.insert = (ele) => {
+    var index = this.heap.length;
+    var arr = [...this.heap];
+    arr.push(ele);
+    while (ele > arr[Math.floor(index / 2)] && index > 1) {
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md:54]:
```suggestion
    while (ele > arr[Math.floor(index / 2)]) {
```

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md [hunk 6]
```diff
@@ -46,6 +46,21 @@ tests:
...
+      arr[index] = arr[Math.floor(index / 2)];
+      arr[Math.floor(index / 2)] = ele;
+      index = arr[Math.floor(index / 2)];
+    }
+    this.heap = arr;
+  }
+  this.print = () => {
+    return this.heap.slice(1);
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md:62]:
```suggestion
    return this.heap;
```

**Review by moT01 — COMMENTED** [review]
Curious why we don't just start the array empty?

**Comment by tywmick** [comment]
@moT01 Oh, that's just based on a tip from the description of [Insert an Element into a Max Heap](https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/insert-an-element-into-a-max-heap) to make the math easier:

> For instance, consider this array representation of a binary min heap:
>
> <code>[ 6, 22, 30, 37, 63, 48, 42, 76 ]</code>
>
> The root node is the first element, 6. Its children are 22 and 30. If we look at the relationship between the array indices of these values, for index i the children are 2 * i + 1 and 2 * i + 2. Similarly, the element at index 0 is the parent of these two children at indices 1 and 2. More generally, we can find the parent of a node at any index with the following: Math.floor( (i - 1) / 2 ). These patterns will hold true as the binary tree grows to any size. Finally, we can make a slight adjustment to make this arithmetic even easier by skipping the first element in the array. Doing this creates the following relationship for any element at a given index i:
>
> Example Array representation:
>
> <code>[ null, 6, 22, 30, 37, 63, 48, 42, 76 ]</code>
>
> An element's left child: i * 2
>
> An element's right child: i * 2 + 1
>
> An element's parent: Math.floor( i / 2 )

We can take out the `null` at index 0—we'll just also have to change the math to find child and parent elements. But either way's cool with me. 👍🏼

**Review by moT01 — DISMISSED** [review]
LGTM @tywmick 🎉

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
We should remove the line requesting the camper to add their `insert` method, if we do this.

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md [hunk 7]
```diff
@@ -47,6 +47,21 @@ tests:
...
+  }
   // change code below this line
   // change code above this line
```
Comment (RandellDawson) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.english.md:66]:
```suggestion
  // change code below this line
  
  // change code above this line
```

**Comment by moT01** [comment]
I added your suggestions @Sky020 & @RandellDawson

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by moT01 — APPROVED** [review]
Thanks for contributing to freeCodeCamp @tywmick 🎉

**Comment by tywmick** [comment]
> We should remove the line requesting the camper to add their `insert` method, if we do this.

Whoops, thanks for catching that one!

**Pull request merged**